### DISCUSSION TITLE
Projeto novo com JS ou TS?

### DISCUSSION THREAD

**Discussion by wilsonneto-swe** [frontendbr/forum#1784] [Perguntas e Respostas]
Fala galera!

Contextualizando: Nossa equipe já fez vários projetos em React com js puro, agora vamos reescrever a aplicação principal, a maior rs Nem todos na equipe manjam de TS mas aprendem fácil se for o caso...

O que vc optaria para um novo projeto, o core da empresa, React com JS ou com TS? Quais as vantagens e desvantagens?

Temos preocupação com mão de obra, contratação, compatibilidade com SSR (Next), produtividade, testes, segurança, engajamento da equipe e etc...

Poderiam me dar opiniões?

Muito obrigado! Abraços!

**Comment by ninetails** [comment]
Pessoalmente, TS.

Mas diria sempre para levar em conta pensar na equipe que vai desenvolver/dar manutenção. Não deixa de ser uma curva acentuada de conhecimento pro projeto e o início vai ter aquela perda de produtividade até adaptar e conhecer melhor. Por outro lado, trabalhar com tipagem traz uma maior segurança a mais em relação ao código desenvolvido. Outro bom ponto é que a comunidade é bem proativa, difícil ter pacote bastante usado sem tipagem no [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped).

**Comment by romulosardinha** [comment]
TS <3
Comecei recentemente e a mão fica coçando para refatorar todos os outros projetos ativos para TS kkkk.. mas ai coloco a cabeça no lugar e controlo minha criança interior =(

**Comment by rafaelsevla** [comment]
Concordo com o @ninetails sobre tudo, mas dou um plus aqui: pode ter problema na produtividade até adaptação mas depois que adaptou, a produtividade duplica em comparação ao JS tradicional

O autocomplete do TS eh simplesmente perfeito... e me arrisco dizer que eh um dos melhores atualmente

E se o time ja tiver grande experiencia com o react usando so o JS, a migração pra TS nao vai ser tao problematica! (digo por experiencia propria hehe)

**Comment by ViniciusGularte** [comment]
Vai de TS, ando desenvolvendo todos meus projetos com typescript, o ganho de produtividade foi bem grande, e a quantidade de refatoração/bugs diminuiu bastante,  concordo com  tudo  o que falaram acima.

**Comment by NegriHenrique** [comment]
vai de TS se vc tiver conhecimento, vale muito mais a pena na minha opinião, para aprender vc terá um pouco de problema com produtividade até o pessoal pegar o jeito mas a longo prazo o TS prevalece muito produtivamente falando

**Comment by rodrigodiasnoronha** [comment]
Acredito que ir direto em um projeto grande com uma equipe que não está muito bem acostumada é um pouco arriscado, mas em projetos pequenos é bom começar a implantar essa cultura de projetos em TS. TS É BOM DEMAIS!!

**Comment by WalyssonPaiva** [comment]
Acredito que caso sua equipe não tenha muita familiaridade com ts, a produtividade pode ser beem reduzida em relação ao js. Mas, a manutenção será bem mais fácil com ts, ainda mais por ser uma aplicação grande, então acredito que compensa

**Comment by wilsonneto-swe** [comment]
Fala pessoal!! 
Muito obrigado pelas dicas ai, 
começamos com TS!!

Já percebemos vantagens muito legais, muito obrigado a todos!!

**Comment by tcelestino** [comment]
No meu caso, eu senti um problema recentemente. Por está usando Typescript há um tempo (alguns meses, para ser mais preciso), precisei reescrever algo usando JS e tive problemas de deixar de lado a "cultura" de uso do Typescript. :laughing:

**Comment by dionarodrigues** [comment]
Depois que se aprende TS, você nunca mais vai querer deixá-lo. 😁
